datafusion/core/tests/sql/joins.rs

```diff
@@ -829,7 +829,13 @@ async fn inner_join_nulls() {
     let sql = "SELECT * FROM (SELECT null AS id1) t1
             INNER JOIN (SELECT null AS id2) t2 ON id1 = id2";
 
-    let expected = vec!["++", "++"];
+    let expected = vec![
```

Comment:
This answer is not correct -- there should be no rows that match.

This is because the join should produce rows where `id1 = id2` evaluates to `true` 

However, `null = null` evaluates to `null` 🤯 

Here is the query in postgres:

```sql
alamb=# SELECT * FROM (SELECT null AS id1) t1
            INNER JOIN (SELECT null AS id2) t2 ON id1 = id2
alamb-# ;
 id1 | id2 
-----+-----
(0 rows)

```